### DISCUSSION TITLE
Is it possible to use Prisma to create new columns / update the schema at runtime?

### DISCUSSION THREAD

**Discussion by kungpaogao** [prisma/prisma#11398] [Q&A] [answered]
I'm thinking about a use case where I want users to be able to define the schema. For example, I would create an app (simple UI) for users to handle their inventory. However, users should be able to define the schema of the items they store in the inventory.

What is a good approach to doing this, and is it possible (or correct) to use Prisma in this case?

I'm wondering this because, from my understanding, Prisma handles schemas statically (have to run `db pull`). I could run a raw SQL query to create new columns at runtime, but then the code base would be out of sync with the actual database.

**Comment by austincrim** [comment, marked as answer]
Hey @kungpaogao 👋 ,

While you mentioned you could add columns with SQL using `prisma.$executeRaw`, I would not recommend this approach. If you have unstructured data that isn't known ahead of time, you could look into using the [`Json` data type](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-fields/working-with-json-fields).

If you have other ideas about how we could solve this, please open a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with more details!

Let me know if you have other questions.

**Reply by vladi-stevanovic** [reply]
👋 Hello @kungpaogao I'm going through pending GitHub Discussions to check if you need any help. Is this issue still open?
Please let us know if we can investigate it further or if you've already solved it, we'd love to know how!

It would be super useful for other users facing this problem! :raised_hands:

**Reply by kungpaogao** [reply]
I didn't end up looking too deeply into this issue, but I think the conclusion is that Prisma is probably not the right tool for this

**Reply by pantharshit00** [reply]
Hello! Thanks for getting back to us. 

Feel free to reaching out again if you choose to use Prisma in any of your other projects.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
